### ISSUE TITLE
Clean up "-D" compile time macros used in pc-linux build

### ISSUE THREAD

**Issue by skliper** [nasa/PSP#21]
The "pc-linux" PSP defines the following to be added to the compiler CFLAGS for __all__ files in the mission:

{{{-D_EL -DENDIAN=_EL -DSOFTWARE_LITTLE_BIT_ORDER -D__ix86__ -D_ix86_ -Dposix -DX86PC -D_REENTRANT -D_EMBED_ -DOS_DEBUG_LEVEL=3}}}

These were brought into the cmake build from the original build scripts in order to be consistent just in case any code required it.  However, they are unnecessary, many are not even used anywhere in CFE/OSAL, and potentially even wrong.  

The reality is that with Linux, the "pc-linux" is a general purpose PSP that can most likely be used on __any__ general-purpose development machine that runs Linux.  It is not limited to only x86 PC's, and in fact works just fine on ARM, PowerPC, and Microblaze targets too.

I have successfully used the (unmodified) pc-linux PSP to execute CFE on a BeagleBone Black (ARM) as well as an emulated PowerPC 440 based development machine.  In all these cases, the "x86" macros are wrong, and on the PowerPC, the EL/ENDIAN/SOFTWARE_LITTLE_BIT_ORDER are wrong too.

To summarize - I recommend removing **ALL** of these macros from the pc-linux build when using the cmake scripts (the old makefiles can stay as-is).

**Comment by skliper** [comment]
_Imported from trac issue 17.  Created by jphickey on 2015-03-18T10:04:45, last modified: 2019-04-02T14:49:34_

**Comment by skliper** [comment]
_Trac comment by jphickey on 2015-04-06 12:02:39:_

This is ready for review/merge

See Commit: [changeset:c6db0f6]

**NOTE**: This changes a script originally added in trac #4 o this is __not__ based off the CFE 4.1.1 seed like the rest of the trac branches are.

**Comment by skliper** [comment]
_Trac comment by glimes on 2015-04-07 14:50:34:_

Changeset [changeset:c6db0f6] looks good for inclusion
in the updated 2015-03-10 merge.

**Comment by skliper** [comment]
_Trac comment by glimes on 2015-04-13 13:50:43:_

Part of integration candidate 2015-03-10,
committed to PSP Development branch on 2015-04-10
as part of merge [changeset:945cb59].

**Comment by skliper** [comment]
_Trac comment by glimes on 2016-01-11 10:42:10:_

Mark for PSP 1.3 per email from sstrege

**Comment by skliper** [comment]
_Trac comment by jhageman on 2019-04-02 14:49:34:_

Milestone renamed
